packages/markers/src/browser/problem/problem-decorator.ts

```diff
@@ -55,10 +56,13 @@ export class ProblemDecorator implements TreeDecorator {
             return result;
         }
         const markers = this.appendContainerMarkers(tree, this.collectMarkers(tree));
-        for (const { id } of new DepthFirstTreeIterator(tree.root)) {
-            const marker = markers.get(id);
-            if (marker) {
-                result.set(id, marker);
+        for (const node of new DepthFirstTreeIterator(tree.root)) {
+            const nodeUri = FileStatNode.getUri(node);
+            if (nodeUri) {
+                const marker = markers.get(nodeUri);
+                if (marker) {
+                    result.set(nodeUri, marker);
```

Comment:
`result.set(node.id, marker);`

Reply:
I changed it back. 
But 'using id' vs 'using uri' here doesn't make difference, does it :) ? At the end the function returns a map where the ids are uris and values are decorators. same logic applies to the git-decorator.

Reply:
> But 'using id' vs 'using uri' here doesn't make difference, does it :) ?

I don't get it, why? It is 2 different things.

Reply:
This function returns 
```ts
return new Map(Array.from(result.values()).map(m => [m.uri, this.toDecorator(m)] as [string, TreeDecoration.Data]));
```
from the above line of code we could tell only `result.values()` goes into the returned value. `result.set(nodeUri, marker);` vs `result.set(node.id, marker);` does not make difference, because `nodeUri` vs 'node.id' only affects the `key` of the `result` map. In the context of problem-decorator and git-decorator, tree nodes with the same uri would have the same decorators.

Reply:
this line is assuming that ID is URI, which is not true in this PR, it is bogus and should be changed as well